python/GafferArnoldUITest/InteractiveArnoldRenderPerformanceTest.py

```diff
@@ -132,72 +132,91 @@ def runInteractive( self, useUI, useBlur, resolution ):
+				with GafferTest.TestRunner.PerformanceScope() as ps:
+					GafferUI.EventLoop.mainEventLoop().start()
+					ps.setNumIterations( GafferImageUI.ImageGadget.tileUpdateCount() )
 
-			script['InteractiveArnoldRender']['state'].setValue( GafferScene.InteractiveRender.State.Stopped )
+				script['InteractiveArnoldRender']['state'].setValue( GafferScene.InteractiveRender.State.Stopped )
 
-			del window, viewer, timer
-			self.waitForIdle( 10 )
+				del window, viewer, timer
+				self.waitForIdle( 10 )
 
-		else:
-			with GafferTest.ParallelAlgoTest.UIThreadCallHandler() as h :
+			else:
+				with GafferTest.ParallelAlgoTest.UIThreadCallHandler() as h :
 
-				with IECore.CapturingMessageHandler() as mh :
-					script['InteractiveArnoldRender']['state'].setValue( GafferScene.InteractiveRender.State.Running )
-					h.waitFor( 2 )
-				arnoldStartupErrors = mh.messages
+					with IECore.CapturingMessageHandler() as mh :
+						script['InteractiveArnoldRender']['state'].setValue( GafferScene.InteractiveRender.State.Running )
+						h.waitFor( 2 )
+					arnoldStartupErrors = mh.messages
 
-				startTime = 0
+					startTime = 0
 
-				tc = Gaffer.ScopedConnection( GafferImageTest.connectProcessTilesToPlugDirtiedSignal( watchNode["out"] ) )
+					tc = Gaffer.ScopedConnection( GafferImageTest.connectProcessTilesToPlugDirtiedSignal( watchNode["out"] ) )
 
-				with GafferTest.TestRunner.PerformanceScope() as ps:
-					with Gaffer.PerformanceMonitor() as m:
-						startTime = time.time()
-						for i in range( 250 ):
-							script["Camera"]["transform"]["translate"]["x"].setValue( math.sin( ( i + 1 ) * 0.1 ) )
-							h.waitFor( 0.02 )
-						endTime = time.time()
+					with GafferTest.TestRunner.PerformanceScope() as ps:
+						with Gaffer.PerformanceMonitor() as m:
+							startTime = time.time()
+							for i in range( 250 ):
+								script["Camera"]["transform"]["translate"]["x"].setValue( math.sin( ( i + 1 ) * 0.1 ) )
+								h.waitFor( 0.02 )
+							endTime = time.time()
 
-					ps.setNumIterations( m.plugStatistics( watchNode["out"]["channelData"].source() ).computeCount )
+						ps.setNumIterations( m.plugStatistics( watchNode["out"]["channelData"].source() ).computeCount )
 
 
-				script['InteractiveArnoldRender']['state'].setValue( GafferScene.InteractiveRender.State.Stopped )
+					script['InteractiveArnoldRender']['state'].setValue( GafferScene.InteractiveRender.State.Stopped )
+
+		finally:
+			IECore.MessageHandler.setDefaultHandler( originalMessageHandler )
 
+		for m in mh.messages:
+			if (
+					m.message.endswith( "rendering with watermarks because ARNOLD_LICENSE_ORDER = none" ) or
+					m.message == "Could not find supported floating point texture format in OpenGL.  GPU image viewer path will be low quality, recommend switching to CPU display transform, or resolving graphics driver issue." or
+					m.message.endswith( "test_profile.json: No such file or directory" )
+					):
+				pass
+			else:
```

Comment:
Is this to avoid `warning` severity messages only? We added a mechanism for this a little while back, which might be a bit more robust:

https://github.com/GafferHQ/gaffer/blob/master/python/GafferArnoldTest/InteractiveArnoldRenderTest.py#L59

Reply:
Cool, that works well.